File: java/src/test/java/problems1_100/TwoSumTest.java
package problems1_100;

import java.util.Arrays;
import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class TwoSumTest extends SampledTest {
    @Test
    public void test() {
        for (TwoSum solution : TwoSum.values()) {
            for (TwoSumSample sample : getSamples(TwoSumSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(
                        solution.twoSum(
                            sample.input.nums,
                            sample.input.target
                        )
                    ).asList()
                    .containsExactlyElementsIn(
                        Arrays
                            .stream(sample.output)
                            .boxed()
                            .toList()
                    ).inOrder();
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems1_100/TwoSumTest.kt
package problems1_100

import com.google.common.truth.Truth.assertWithMessage
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class TwoSumTest : SampledTest() {
    @Test
    fun test() =
        TwoSum.entries.forEach { solution ->
            getSamples<Array<TwoSumSample>>().forEach { (input, output, message) ->
                assertWithMessage(message)
                    .that(solution.twoSum(input.nums, input.target))
                    .asList()
                    .containsExactlyElementsIn(output.toList())
                    .inOrder()
            }
        }
}
